src/main.dart

```diff
@@ -1,8 +1,99 @@
-// TODO: Run task list app
-void main(List<String> args) {
-  print('Hello World');
+import 'dart:developer';
+
+import 'dart:io';
+
+class Task {
+  int? taskID;
+  String? name;
+  String? description;
+  bool? isCompleted;
+
+  Task(this.taskID, this.name, this.description, this.isCompleted);
+}
+
+class TaskManager {
+  List<Task> tasks = [];
+  int taskID = 1;
+
+  // Add task
+  void addTask(String? name, String? description) {
+    tasks.add(Task(taskID, name, description, false));
+    taskID++;
+  }
+
+  // List Tasks
+  void listTask() {
+    for (int i = 0; i < tasks.length; i++) {
+      Task task = tasks[i];
+      print(
+          "${task.taskID}. [${task.isCompleted! ? 'X' : ' '}] ${task.name}: ${task.description}");
```

Comment:
while this is okay, maybe design a better way to print all tasks. This is just an example but would make description more clear and separate tasks more clearly
```suggestion
          print("${task.taskID}. [${task.isCompleted! ? 'X' : ' '}] ${task.name}\nDescription: ${task.description}");
          print("=============================================================================================");
```